tests/gun_aiming.cpp

```diff
@@ -8,19 +8,16 @@ static void test_internal( const npc& who, const item &gun )
 {
     THEN( "the effective range is correctly calcuated" ) {
         // calculate range for 50% chance of critical hit at arbitrary recoil
-        double recoil = rng_float( 0, 1000 );
-        double range = who.gun_current_range( gun, recoil, 50, accuracy_critical );
+        for ( double recoil = 0; recoil < 1000; recoil += 100 ) {
+            double range = who.gun_current_range( gun, recoil, 50, accuracy_critical );
+            double dispersion = who.get_weapon_dispersion( gun ) + recoil;
 
-        // calculate actual accuracy at the given range
-        double dispersion = ( who.get_weapon_dispersion( gun ) + recoil ) / 2;
-        double missed_by = iso_tangent( range, dispersion );
+            CAPTURE( recoil );
+            CAPTURE( range );
+            CAPTURE( dispersion );
 
-        INFO( "Recoil: " << recoil );
-        INFO( "Range: " << range );
-        INFO( "Dispersion: " << dispersion );
-
-        // require inverse calculation to agree with tolerance of 0.1%
-        REQUIRE( std::abs( missed_by - accuracy_critical ) < accuracy_critical / 1000 );
+            REQUIRE( who.projectile_attack_chance( dispersion, range, accuracy_critical ) == Approx( 0.5 ).epsilon( 0.0005 ) );
```

Comment:
Dont forget to update `accuracy_critical` here or even better define both this and the chance as separate variables and then iterate through different values for chance in an outer loop.

I suspect your implementation would pass such a test whereas your comment with the math reasoning in it suggests the current implementation would not (would fail for `chance != 50`)